app/models/warehouse/build_report.rb

```diff
@@ -138,6 +139,13 @@ def fill_cas_report_table!
                 end
               end
 
+              # If this a canceled or declined decision, log who did it
+              event_contact = if decision.status.in?(['declined', 'canceled'])
+                decision.decision_action_events.select do |da|
```

Comment:
Use `detect` instead of `select` so you don't need the `first`?

Reply:
Of course.  Thanks